src/Domain/Services/CreateAccount.php

```diff
@@ -120,18 +133,26 @@ public function from_order_request( \WC_Order $order, \WP_REST_Request $request
 	 *
 	 * @return boolean True if a new user account should be created.
 	 */
-	protected function should_create_customer_account( $request ) {
+	protected function should_create_customer_account( \WP_REST_Request $request ) {
 		if ( is_user_logged_in() ) {
+			// User is already logged in - no need to create an account.
 			return false;
 		}
 
-		$checkout_requires_account = false === filter_var( get_option( 'woocommerce_enable_guest_checkout' ), FILTER_VALIDATE_BOOLEAN );
-		if ( $checkout_requires_account ) {
+		// From here we know that the shopper is not logged in.
+
+		if ( false === filter_var( get_option( 'woocommerce_enable_guest_checkout' ), FILTER_VALIDATE_BOOLEAN ) ) {
+			// Store requires an account for all checkouts (purchases).
+			// Create an account independent of shopper option in $request.
+			// Note - checkbox is not displayed to shopper in this case.
 			return true;
 		}
```

Comment:
Incidentally, this is why I think there's value in writing phpunit tests for this service class. Tests help describe expectations for the things you are testing and also help catch changes in those expectations when refactoring happens over the course of a review. I feel we've established a baseline of the approach we're taking where there's not much risk of effort spent on tests being wasted.

Reply:
I agree - I'll get some test coverage in place.

Regarding the relationship to guest account option – I haven't considered not implementing that. It might be fine/possible to simplify it, so I'll give this a try. 

If we shipped it like this it would be different behaviour from existing checkout; though perhaps that's a useful improvement (as you suggest). We might need to document this and/or tweak the UX to avoid surprises/bug reports. FYI @LevinMedia for awareness, I'll follow up when I've investigated further (and/or prototyped it).